design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,202 @@
+---
+title: Visual Query Builder
+description: 
+published: true
+date: 2022-03-08 18:02:53
+tags: 
+editor: markdown
+dateCreated: 2022-03-08 18:02:58
+---
+
+## Context
+
+This document specifies the design for a visual query builder named `Data Explorer` within the Mathesar application. Through using `Data Explorer`, users will be able to analyze data across one or multiple tables of a schema and save the result of those queries as views.
+
+Additionally, users will be able to open existing views in `Data Explorer`, as long as the query commands and functions used to generate it are supported.
+
+For a more detailed overview of this feature, read [Views Product Spec](https://wiki.mathesar.org/en/product/specs/2022-01-views).
+
+## Scenarios
```

Comment:
We are missing scenarios for navigating to and navigating away from the data explorer (from the usual view with tabs and left sidebar).